cmd/proxy/main.go

```diff
@@ -202,24 +219,48 @@ func (o *ProxyRunOptions) Validate() error {
 	if o.adminPort < 1024 {
 		return fmt.Errorf("please do not try to use reserved port %d for the admin port", o.adminPort)
 	}
+
+	// validate agent authentication params
+	// all 4 parametes must be empty or must have value (except kubeconfigPath that might be empty)
+	if o.agentNamespace != "" || o.agentServiceAccount != "" || o.authenticationAudience != "" || o.kubeconfigPath != "" {
+		if o.agentNamespace == "" {
+			return fmt.Errorf("agentNamespace cannot be empty when agent authentication is enabled")
```

Comment:
For future we should consider accumulating these errors. Sort of annoying to be given and error I need a service account on run1 and then be given an error I need an audience on run2.